### PR TITLE
Feature timeline fabric

### PR DESCRIPTION
**Pull request by cadw**
Timeline is now drawn using FabricJs, only problem is that we do not currently have the 'bar' attribute for cues, so after we add this, we can easily draw the cues on the timeline. Howeverm I think we van merge this already, since the timeline is drawn without problems. Features of the timeline:
- Grid based on parametric interval
- Automatically resize canvas to the size of the parent, without losing any quality

### CODE REVIEW

File: src/main/webapp/app/timeline/timeline.controller.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
+(function() {
+    'use strict';
+
+    angular
+        .module('thefirstorderApp')
+        .controller('TimelineController', TimelineController);
+
+    TimelineController.$inject = ['$scope', '$state', 'Cue', 'AlertService'];
+
+    function TimelineController ($scope, $state, Cue, AlertService) {
```
Comment (rubenwiersma) [inline src/main/webapp/app/timeline/timeline.controller.js]: Would it be good to have commentary like java in here?

File: src/main/webapp/app/timeline/timeline.controller.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            }
+            function onError(error) {
+                AlertService.error(error.data.message);
+            }
+        }
+
+        function drawTimeline(cues) {
+            var htmlCanvas = document.getElementById('concertTimeline');
+
+            // Resize the timeline to the size of the parent
```
Comment (rubenwiersma) [inline src/main/webapp/app/timeline/timeline.controller.js]: This comment isn't necessary, the name of the function is descriptive enough.

File: src/main/webapp/app/timeline/timeline.html [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<div ng-cloak>
+    <div class="row">
+        <div class="col-md-12">
+            <h2 class="text-center">Timeline</h2><br /><br />
+            <div id="timeline">
+                <canvas id="concertTimeline" width="1200" height="50"></canvas>
```
Comment (rubenwiersma) [inline src/main/webapp/app/timeline/timeline.html]: Is this hard-coded width and height still necessary with the new fit functionality?

**Comment by rubenwiersma** [comment]
Overall: good code, some comments missing and some redundant.

**Comment by hungnguyen94** [comment]
Looks good. You should also add proper documentation for the AngularJS code though. Look into ngdocs.

**Pull request merged**